flutter/shell/platform/tizen/nui_autofill_popup.cc

```diff
@@ -0,0 +1,86 @@
+
+bool NuiAutofillPopup::Touched(Dali::Actor actor,
+                               const Dali::TouchEvent& event) {
+  const Dali::PointState::Type state = event.GetState(0);
+  if (Dali::PointState::DOWN == state) {
+    std::string text =
+        actor.GetProperty(Dali::Actor::Property::NAME).Get<std::string>();
+    on_commit_(text);
+    popup_.SetDisplayState(Dali::Toolkit::Popup::HIDDEN);
+  }
+  return true;
+}
+
+void NuiAutofillPopup::Hidden() {
+  // TODO(Swanseo0): There is a phenomenon where white traces remain for a
+  // while when popup disappears.
+  popup_.Unparent();
+  popup_.Reset();
+}
+
+void NuiAutofillPopup::OutsideTouched() {
+  popup_.SetDisplayState(Dali::Toolkit::Popup::HIDDEN);
+}
+
+void NuiAutofillPopup::Prepare() {
+  popup_ = Dali::Toolkit::Popup::New();
+  popup_.SetProperty(Dali::Actor::Property::NAME, "popup");
+  popup_.SetProperty(Dali::Actor::Property::PARENT_ORIGIN,
+                     Dali::ParentOrigin::TOP_LEFT);
+  popup_.SetProperty(Dali::Actor::Property::ANCHOR_POINT,
+                     Dali::AnchorPoint::TOP_LEFT);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::TAIL_VISIBILITY, false);
+  popup_.SetBackgroundColor(Dali::Color::WHITE_SMOKE);
+  popup_.OutsideTouchedSignal().Connect(this,
+                                        &NuiAutofillPopup::OutsideTouched);
+  popup_.HiddenSignal().Connect(this, &NuiAutofillPopup::Hidden);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::BACKING_ENABLED, false);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::AUTO_HIDE_DELAY, 2500);
+}
+
+void NuiAutofillPopup::Show(Dali::Actor* actor) {
+  const std::vector<std::unique_ptr<AutofillItem>>& items =
+      TizenAutofill::GetInstance().GetResponseItems();
+  if (!items.empty()) {
+    Prepare();
+    Dali::Toolkit::TableView content =
+        Dali::Toolkit::TableView::New(items.size(), 1);
+    content.SetResizePolicy(Dali::ResizePolicy::FILL_TO_PARENT,
+                            Dali::Dimension::ALL_DIMENSIONS);
+    content.SetProperty(Dali::Actor::Property::PADDING,
+                        Dali::Vector4(10, 10, 0, 0));
+    for (uint32_t i = 0; i < items.size(); ++i) {
+      Dali::Toolkit::TextLabel label =
+          Dali::Toolkit::TextLabel::New(items[i]->label_);
+      label.SetProperty(Dali::Actor::Property::NAME, items[i]->value_);
+      label.SetResizePolicy(Dali::ResizePolicy::DIMENSION_DEPENDENCY,
+                            Dali::Dimension::HEIGHT);
+      label.SetProperty(Dali::Toolkit::TextLabel::Property::TEXT_COLOR,
+                        Dali::Color::WHITE_SMOKE);
+      label.SetProperty(Dali::Toolkit::TextLabel::Property::POINT_SIZE, 7.0f);
+      label.TouchedSignal().Connect(this, &NuiAutofillPopup::Touched);
+      content.AddChild(label, Dali::Toolkit::TableView::CellPosition(i, 0));
+      content.SetFitHeight(i);
+    }
+    popup_.SetProperty(Dali::Actor::Property::SIZE,
+                       Dali::Vector2(140.0f, 35.0f * items.size()));
```

Comment:
The more items, the longer the popup.
I think it should be scrollable after a certain number, what do you think?

Reply:
> Does "white background" and "7.0f font size" look good for both tv and common (rpi) profiles?

Reducing the size was better than the default.

> +) NUI popup may differ from Dali popup. It might be better to set it equal to that value.

In order to make Dali's popup the same as NUI's popup, I think we need to find a way to set Property in more detail. I tried to make it the same shape, but I couldn't find a way to set it up, such as the shape of popup, the border of each item, and etc.

> I think it should be scrollable after a certain number, what do you think?

I agree. I tried to make it scrollable, but... I couldn't find a way.